newIDE/app/src/MainFrame/index.js

```diff
@@ -863,13 +890,23 @@ class MainFrame extends React.Component<Props, State> {
     options: PreviewOptions
   ) => {
     const { _previewLauncher } = this;
+    const {
+      currentProject,
+      previewFirstSceneName,
+      isPreviewOverride,
+    } = this.state;
+
     if (!_previewLauncher) return;
 
     this.setState(
       {
         previewLoading: true,
       },
       () => {
+        if (previewFirstSceneName && isPreviewOverride) {
+          layout = currentProject.getLayout(previewFirstSceneName);
```

Comment:
In general avoid to change parameters. Declare a new variable and use it. The layout passed as parameter is not supposed to be changed to another layout - this could surprise other usage of the layout by the rest of the code and create weird bugs.